app.js

```diff
@@ -20,115 +20,124 @@ var yaml = require('js-yaml');
  */
 function initApp(options) {
 
-    // the main application object
-    var app = express();
-
-    // get the options and make them available in the app
-    app.logger = options.logger;    // the logging device
-    app.metrics = options.metrics;  // the metrics
-    app.conf = options.config;      // this app's config options
-    app.info = packageInfo;         // this app's package info
-
-    // ensure some sane defaults
-    if(!app.conf.port) { app.conf.port = 8888; }
-    if(!app.conf.interface) { app.conf.interface = '0.0.0.0'; }
-    if(app.conf.compression_level === undefined) { app.conf.compression_level = 3; }
-    if(app.conf.cors === undefined) { app.conf.cors = '*'; }
-    if(app.conf.csp === undefined) {
-        app.conf.csp =
-            "default-src 'self'; object-src 'none'; media-src *; img-src *; style-src *; frame-ancestors 'self'";
-    }
-
-    // set outgoing proxy
-    if(app.conf.proxy) {
-        process.env.HTTP_PROXY = app.conf.proxy;
-        // if there is a list of domains which should
-        // not be proxied, set it
-        if(app.conf.no_proxy_list) {
-            if(Array.isArray(app.conf.no_proxy_list)) {
-                process.env.NO_PROXY = app.conf.no_proxy_list.join(',');
-            } else {
-                process.env.NO_PROXY = app.conf.no_proxy_list;
+    return BBPromise.try(() => {
```

Comment:
To avoid a gigantic diff that will not be easy to merge into the services we can leave the initApp function internals untouched and just do `const initApp = BBPromise.method(() => ...` in the function initialization. That one-liner will have the same effect as this huge diff.